### ISSUE TITLE
FR: Make Taildrop usable with tagged devices

### ISSUE THREAD

**Issue by KaryptoX** [tailscale/tailscale#10695]
### What are you trying to do?

Unfortunately taildrop (an ultra awesome feature) is not usable with tagged devices. If an device is tagged it will not be shown in taildrop anymore. 
taildrop is only usable for untagged user devices. 

Would be nice if tagged device support for taildrop could be added in future versions. 

Device Tagging is an essential feature for me, as it makes the acl much clearer and easier to maintain.

### How should we solve this?

Add tagged device support for taildrop feature

### What is the impact of not solving this?

Taildrop is useless for tagged devices

### Anything else?

Tailscale is awesome!!

**Comment by glidingthrough** [comment]
Agreed. I've tagged all machines to restrict access within the tailnet and now Taildrop doesn't work.

This FR would be even better if we could restrict Taildrop access using tags through ACLs (e.g. laptops and mobiles can all send data via taildrop, but servers cannot). Very useful as an extra line of defense when there are internet facing servers.

**Comment by rageshkrishna** [comment]
I'm _very_ interested in seeing this, too. I used to love this feature, but I had to give it up once I tagged my devices.

Is there an explanation somewhere why this doesn't work on tagged devices? Is there a technical limitation that makes it hard or impossible, or is it just a case of prioritization?

**Comment by gnur** [comment]
> Is there an explanation somewhere why this doesn't work on tagged devices? Is there a technical limitation that makes it hard or impossible, or is it just a case of prioritization?

Tagged devices seem to be thought of as "servers" and no longer as "users". They also lose the "identity" of the user that created them. It seems that this might be the reason why taildrop doesn't work on those devices.

**Comment by Tyri0n** [comment]
what a stupid, seemingly pointless, restriction

**Comment by cesaryuan** [comment]
@bradfitz Hi, sorry to bother you. Just want to know if there is an exactly date to provide this feature.

**Comment by meanmrgreen** [comment]
Make it so!

**Comment by augusttty** [comment]
Tailscale restrict ssh to a tagged resouce with ACL configured, so one must tagged a server for using ssh with magicDNS.But then the taildrop doesn't work.
It will be great that we can use taildrop on a tagged server: )

**Comment by Mrono** [comment]
What's the point of having taildrop if the use of another major feature precludes you from using taildrop...

**Comment by hevel86** [comment]
This was opened in 2023, it's the middle of 2025.  Is there any place to follow the progress of taildrop + tagged devices, specifically?

**Comment by ws-mwayne** [comment]
Also interested in this...

**Comment by noidwasavailable** [comment]
Is there any update on this issue? I would really like to use Taildrop between my non-macos devices :D

**Comment by kylesmithit** [comment]
Also very interested in this.

**Comment by patricksthannon** [comment]
Moved to tagged devices and would like to use this feature still. Bummer this is an issue.

**Comment by evanpurkhiser** [comment]
Also recently tagged my devices only to find out I can’t taildrop too them now! Just leaving a comment to add to the comment count in hopes this can get some attention :)

**Comment by arduoi** [comment]
https://www.reddit.com/r/Tailscale/comments/1imbq9b/taildrop_to_tagged_devices_or_devices_you_dont/mc35moy/ - come on guys! You can do it

**Comment by tailNAS** [comment]
just recently tagged all my devices to use ACL and can't use Taildrop anymore... I see that this issue has been open in 2023, is there any progress since then?

**Comment by hevel86** [comment]
> just recently tagged all my devices to use ACL and can't use Taildrop anymore... I see that this issue has been open in 2023, is there any progress since then?

@tailNAS if there has been any progress I haven't seen it.  However, an alternative I've used is [LocalSend](https://localsend.org/) with your Tailscale IP.  
For all intents and purposes it does the same thing.
